python/test/conftest.py

```diff
@@ -26,16 +29,141 @@ def eyes_runner_class():
     return None
 
 
+@pytest.fixture(scope="function")
+def options():
+    return webdriver.ChromeOptions()
+
+
+@pytest.fixture(scope="function")
+def browser_type():
+    return "Chrome"
+
+
+@pytest.fixture(scope="function")
+def legacy():
+    return False
+
+
+@pytest.fixture(scope="function")
+def desired_caps():
+    return None
+
+
+@pytest.yield_fixture(scope="function")
+def android_desired_capabilities(request, dev, app):
+    desired_caps = copy(getattr(request, "param", {}))  # browser_config.copy()
+    desired_caps["app"] = app
+    desired_caps["NATIVE_APP"] = True
+    desired_caps["browserName"] = ""
+    desired_caps["deviceName"] = "Samsung Galaxy S8 WQHD GoogleAPI Emulator"
+    desired_caps["platformVersion"] = "8.1"
+    desired_caps["platformName"] = "Android"
+    desired_caps["clearSystemFiles"] = True
+    desired_caps["noReset"] = True
+    desired_caps["name"] = "AndroidNativeApp checkWindow"
+    return desired_caps
+
+
+@pytest.yield_fixture(scope="function")
+def ios_desired_capabilities(request, dev, app):
+    desired_caps = copy(getattr(request, "param", {}))
+    desired_caps["app"] = app
+    desired_caps["NATIVE_APP"] = True
+    desired_caps["browserName"] = ""
+    desired_caps["deviceName"] = "iPhone XS Simulator"
+    desired_caps["platformVersion"] = "12.2"
+    desired_caps["platformName"] = "iOS"
+    desired_caps["clearSystemFiles"] = True
+    desired_caps["noReset"] = True
+    desired_caps["name"] = "iOSNativeApp checkWindow"
+    return desired_caps
+
+
 @pytest.fixture(name="driver", scope="function")
-def driver_setup():
-    options = webdriver.ChromeOptions()
-    options.add_argument("--headless")
+def driver_setup(options, browser_type, desired_caps):
+    #options = webdriver.ChromeOptions()
     counter = 0
+    sauce_url = (
+            "https://{username}:{password}@ondemand.saucelabs.com:443/wd/hub".format(
+                username=os.getenv("SAUCE_USERNAME", None),
+                password=os.getenv("SAUCE_ACCESS_KEY", None),
+            )
+        )
     while counter < 5:
```

Comment:
instead of `while counter < 5:` you could use `for _ in range(5)`. This would remove the `counter` variable.